### PR TITLE
Addnews

### PR DESCRIPTION
**Pull request by shahab-ramzan**
added dropdown for nick name selection and only the author will be able to delete the news

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
     images: {
-      domains: ["api3.canonizer.com"],
+      domains: ["api3.canonizer.com","api3.canonizer.comfa fa-facebook"],
```
Comment (akuks) [inline next.config.js]: There is no space in the second element of domains array. Is it correct?

File: src/components/ComponentPages/News/Edit/index.tsx [hunk 2]
```diff
@@ -1,154 +1,139 @@
...
-      id: values.data.map((id) => id.id),
-      display_text: values.data.map((text) => text.display_text),
-      link: values.data.map((link) => link.link),
-      available_for_child: values.data.map(
-        (available) => available.available_for_child
-      ),
-    };
-    const res = await updateNewsDataApi(dataObj);
+    const res = await updateNewsFeedApi({
+      newsfeed_id: dataToUpdate?.id,
```
Comment (akuks) [inline src/components/ComponentPages/News/Edit/index.tsx]: Showing an error `Property 'id' does not exist on type`

File: src/components/ComponentPages/News/Edit/index.tsx [hunk 3]
```diff
@@ -1,154 +1,139 @@
...
+      display_text: values.display_text,
+      link: values.link,
+      available_for_child: values.available_for_child,
+      submitter_nick_id: dataToUpdate?.submitter_nick_id,
```
Comment (akuks) [inline src/components/ComponentPages/News/Edit/index.tsx]: Showing error **`Property 'submitter_nick_id' does not exist on type '{}'.`**

File: src/components/ComponentPages/News/Edit/index.tsx [hunk 4]
```diff
@@ -1,154 +1,139 @@
...
-                  <>
-                    <Card
-                      className={styles.newsFormFieldsCard}
-                      bordered={false}
-                      key={field.key}
-                    >
-                      <Badge className={styles.newsFormFieldsCardCounter}>
-                        {field.key}
-                      </Badge>
+      <Form form={form} name="basic" layout={"vertical"} onFinish={onFinish}>
```
Comment (akuks) [inline src/components/ComponentPages/News/Edit/index.tsx]: I feel we can combine the form of Add News and Edit News. Both almost have the same fields

File: src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx [hunk 5]
```diff
@@ -41,23 +48,48 @@ const NewsFeedsCard = ({ newsFeed }) => {
...
+                      onClick={() => {
+                        if (isLogin) {
+                          setEditNews(true);
+                          setDeleteNews(false);
+                        } else {
+                          router.push("/login");
+                        }
+                      }}
+                    >
                       <i className={"icon-edit "}></i>Edit News
```
Comment (akuks) [inline src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx]: Probably we can collapse the empty tags

File: src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx [hunk 6]
```diff
@@ -41,23 +48,48 @@ const NewsFeedsCard = ({ newsFeed }) => {
...
+                      onClick={() => {
+                        if (isLogin) {
+                          setDeleteNews(true);
+                          setEditNews(false);
+                        } else {
+                          router.push("/login");
+                        }
+                      }}
+                    >
+                      <i className={"icon-delete"}></i>Delete News
```
Comment (akuks) [inline src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx]: Collapse the empty tags

File: src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx [hunk 7]
```diff
@@ -68,16 +100,56 @@ const NewsFeedsCard = ({ newsFeed }) => {
...
-                        <DeleteOutlined
-                          onClick={() => handleDeleteCamp(news.id)}
-                        />
+                        <Tooltip
+                          title={news.owner_flag ? "" : "must owner to delete"}
+                        >
+                          <Popconfirm
+                            disabled={!news.owner_flag}
+                            placement="topLeft"
+                            title="Are you sure to delete this task?"
```
Comment (akuks) [inline src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx]: If we can move this message to config file. And `Are you sure you want to delete this news?`

File: src/components/common/editor/index.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
+import dynamic from "next/dynamic";
+import ReactQuill from "react-quill";
+import React from "react";
+const QuillNoSSRWrapper = dynamic(() => ReactQuill, { ssr: false });
```
Comment (akuks) [inline src/components/common/editor/index.tsx:4]: Not sure, why are we using Quill editor.

**Review by akuks — CHANGES_REQUESTED** [review]

File: next.config.js (hunk 1 above)
Comment (shahab-ramzan) [reply next.config.js]: removed

File: src/components/ComponentPages/News/Edit/index.tsx (hunk 2 above)
Comment (shahab-ramzan) [reply src/components/ComponentPages/News/Edit/index.tsx]: fixed

File: src/components/ComponentPages/News/Edit/index.tsx (hunk 4 above)
Comment (shahab-ramzan) [reply src/components/ComponentPages/News/Edit/index.tsx]: refactored

File: src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx (hunk 5 above)
Comment (shahab-ramzan) [reply src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx]: Done

File: src/components/ComponentPages/News/Edit/index.tsx (hunk 3 above)
Comment (shahab-ramzan) [reply src/components/ComponentPages/News/Edit/index.tsx]: fixed

File: src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx (hunk 6 above)
Comment (shahab-ramzan) [reply src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx]: Done

File: src/components/common/editor/index.tsx (hunk 8 above)
Comment (shahab-ramzan) [reply src/components/common/editor/index.tsx:4]: This one is added by Rahul for text editing experiments, i guess

File: src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx (hunk 7 above)
Comment (shahab-ramzan) [reply src/components/ComponentPages/TopicDetails/NewsFeedsCard/index.tsx]: I have created a ticket for next sprint in which I will set up config file and move all the exceptional messages to config file

**Pull request merged**